source/grid/grid_generator_pipe_junction.cc

```diff
@@ -69,6 +69,194 @@ namespace
+       * the distance from the axis, $\phi$ the angle between the given point
+       * and the normal direction, and $\lambda$ the axial position.
+       */
+      virtual Point<3>
+      pull_back(const Point<spacedim> &space_point) const override;
+
+      /**
+       * Compute the Cartesian coordinates for a chart point given in
+       * cylindrical coordinates $(r, \phi, \lambda)$ on a cylinder of height
+       * one, where $r$ denotes the distance from the axis, $\phi$ the angle
+       * between the given point and the normal direction, and $\lambda$ the
+       * axial position.
+       */
+      virtual Point<spacedim>
+      push_forward(const Point<3> &chart_point) const override;
+
+    protected:
+      /**
+       * A vector orthogonal to the normal direction.
+       */
+      const Tensor<1, spacedim> normal_direction;
+
+      /**
+       * The direction vector of the axis.
+       */
+      const Tensor<1, spacedim> direction;
+
+      /**
+       * An arbitrary point on the axis.
+       */
+      const Point<spacedim> point_on_axis;
+
+    private:
+      /**
+       * Pipe segment properties to calculate its height.
+       */
+      const AdditionalData data;
+
+      /**
+       * Relative tolerance to measure zero distances.
+       */
+      const double tolerance;
+
+      /**
+       * The direction vector perpendicular to both direction and
+       * normal_direction.
+       */
+      const Tensor<1, spacedim> dxn;
+    };
+
+
+
+    template <int dim, int spacedim>
+    Manifold<dim, spacedim>::Manifold(
+      const Tensor<1, spacedim> &normal_direction,
+      const Tensor<1, spacedim> &direction,
+      const Point<spacedim> &    point_on_axis,
+      const AdditionalData &     data,
+      const double               tolerance)
+      : ChartManifold<dim, spacedim, 3>(Tensor<1, 3>({0, 2. * numbers::PI, 0}))
+      , normal_direction(normal_direction)
+      , direction(direction)
+      , point_on_axis(point_on_axis)
+      , data(data)
+      , tolerance(tolerance)
+      , dxn(cross_product_3d(direction, normal_direction))
+    {
+      Assert(spacedim == 3,
+             ExcMessage(
+               "PipeSegment::Manifold can only be used for spacedim==3!"));
+
+      Assert(std::abs(normal_direction.norm() - 1) < tolerance,
+             ExcMessage("Normal direction must be unit vector."));
+      Assert(std::abs(direction.norm() - 1) < tolerance,
+             ExcMessage("Direction must be unit vector."));
+      Assert(normal_direction * direction < tolerance,
+             ExcMessage(
+               "Direction and normal direction must be perpendicular."));
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::unique_ptr<dealii::Manifold<dim, spacedim>>
+    Manifold<dim, spacedim>::clone() const
+    {
+      return std::make_unique<Manifold<dim, spacedim>>(
+        normal_direction, direction, point_on_axis, data, tolerance);
```

Comment:
```suggestion
      return std::make_unique<Manifold<dim, spacedim>>(*this);
```
:-)

Reply:
Ah, this is nice.

Might be a good idea to update the `clone` function from the other `Manifold` classes in `grid/manifold_lib.cc` as well.

Reply:
See https://github.com/dealii/dealii/pull/13307.